tools/shared.py

```diff
@@ -3249,6 +3249,9 @@ def read_and_preprocess(filename, expand_macros=False):
   run_js(path_from_root('tools/preprocessor.js'), NODE_JS, args, True, stdout=open(stdout, 'w'), cwd=path)
   out = open(stdout, 'r').read()
 
+  os.remove(stdout)
```

Comment:
these should not be needed, as the temp dir is cleaned up - did you see a bug there?

Reply:
I had test_debuginfo telling me that 11 temp files were left over in one of the test runs, which I couldn't reproduce locally, so I put this in to see if it helped.  On reflection, I don't think it did anything and the issue went away following a rebase.  I'll take these lines out.